nemo/core/neural_types/axes.py

```diff
@@ -0,0 +1,77 @@
+# ! /usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = ['AxisKindAbstract', 'AxisKind', 'AxisType']
+
+from enum import Enum
+from typing import Optional
+
+
+class AxisKindAbstract(Enum):
+    """This is an abstract Enum to represents what does varying axis dimension mean.
+    In practice, you will almost always use AxisKind Enum. This Enum should be inherited by
+    your OWN Enum if you aren't satisfied with AxisKind. Then your own Enum can be used
+    instead of AxisKind."""
+
+    pass
+
+
+class AxisKind(AxisKindAbstract):
+    """This Enum represents what does varying axis dimension mean.
+    For example, does this dimension correspond to width, batch, time, etc."""
+
+    Batch = 0
+    Time = 1
+    Dimension = 2
```

Comment:
Can we change that name to "Channel"? 

The word _dimension_ is equivalent to our _axis_ (number of tensor dimensions = number of axes = tensor rank)

Reply:
Dimension seems more general to me - I only hear about "Channel" in the context of CNNs.
However, the shorthand notation "C" and "D" will resolve to the same thing, so:
('B', 'T', 'C') and ('B', 'T', 'D') are the same things